File: src/main/java/fi/aalto/cs/apluscourses/ui/DuplicateSubmissionDialog.java
package fi.aalto.cs.apluscourses.ui;

import static fi.aalto.cs.apluscourses.utils.PluginResourceBundle.getText;

import com.intellij.openapi.ui.Messages;

public class DuplicateSubmissionDialog {

  private DuplicateSubmissionDialog() {

  }

  /**
   * Displays a dialog asking the user whether they really want to submit a duplicate submissions.
   * @return True if the user wants to proceed with the submission.
   */
  public static boolean showDialog() {
    final String[] options = { getText("ui.duplicateDialog.yesOption"), getText("ui.duplicateDialog.noOption") };

    return Messages.showDialog(getText("ui.duplicateDialog.content"),
        getText("ui.duplicateDialog.title"),
        options,
        1,
        null) == 0;
  }
}

File: src/main/kotlin/fi/aalto/cs/apluscourses/ui/exercise/DuplicateSubmissionDialog.kt
package fi.aalto.cs.apluscourses.ui.exercise

import com.intellij.openapi.ui.Messages
import fi.aalto.cs.apluscourses.MyBundle.message

object DuplicateSubmissionDialog {
    // TODO check if code has been compiled
    /**
     * Displays a dialog asking the user whether they really want to submit a duplicate submissions.
     *
     * @return True if the user wants to proceed with the submission.
     */
    fun showDialog(): Boolean {
        val options =
            arrayOf<String?>(message("ui.duplicateDialog.yesOption"), message("ui.duplicateDialog.noOption"))

        return Messages.showDialog(
            message("ui.duplicateDialog.content"),
            message("ui.duplicateDialog.title"),
            options,
            1,
            null
        ) == 0
    }
}
